presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/CommandExecutor.java

```diff
@@ -13,26 +13,53 @@
  */
 package com.teradata.prestomanager.agent;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.File;
+import java.util.StringTokenizer;
 import java.util.concurrent.TimeUnit;
 
-import static java.lang.Runtime.getRuntime;
+import static java.lang.ProcessBuilder.Redirect.appendTo;
 import static java.lang.String.format;
 
-// TODO: Add logger
 public final class CommandExecutor
 {
+    private static final Logger LOGGER = LogManager.getLogger(CommandExecutor.class);
+    private static final File logFile = new File("HttpRequest.log");
+
     private CommandExecutor() {}
 
     public static int executeCommand(String command)
             throws PrestoManagerException
     {
+        LOGGER.debug("Command to be executed: {}", command);
+
+        StringTokenizer tokenizer = new StringTokenizer(command);
+        String[] commandArray = new String[tokenizer.countTokens()];
+        for (int i = 0; tokenizer.hasMoreTokens(); i++) {
+            commandArray[i] = tokenizer.nextToken();
+        }
+
+        ProcessBuilder processBuilder = new ProcessBuilder(commandArray);
+        processBuilder.redirectErrorStream(true);
+        processBuilder.redirectOutput(appendTo(logFile));
+        Process process = null;
         try {
-            Process process = getRuntime().exec(command);
+            process = processBuilder.start();
```

Comment:
instead of the try/catch, do:
```
process = processBuilder.start()
if(!process.waitFor(2, TimeUnit.MINUTES)) {
     process.destroyForcibly();
}
return process.exitValue();
```